src/actions/pagerduty/pagerduty_form.ts

```diff
@@ -0,0 +1,100 @@
+import * as Hub from '../../hub'
+import { PagerDutyCreateIncident } from './pagerduty_create_incident'
+import * as moment from 'moment'
+
+export class PagerDutyForm {
+    static ServiceKeyKey = 'service_key'
+    static DeDeuplicationIdColumnKey = 'deduplication_id_column'
+    static AutoResolveKey = 'auto_resolution_boolean'
+    static MaxIncidentsKey = 'max_incidents'
+
+    static formFields: Hub.ActionFormField[] = [
+        {
+            label: 'Service key',
+            type: 'string',
+            name: PagerDutyForm.ServiceKeyKey,
+            description: 'An Events API v2 integration key',
+            required: true
+        }, {
+            label: 'De-duplication id column',
+            name: PagerDutyForm.DeDeuplicationIdColumnKey,
+            description: 'Column in the result set that uniquely identifies an alert',
+            type: 'string',
+            required: true
+        }
+        , {
+            label: 'Enable automatic resolution',
+            name: PagerDutyForm.AutoResolveKey,
+            description: 'Automatically resolves incidents',
+            type: 'string',
+            default: 'true',
+            required: true
+        },
+        {
+            label: 'Maximum number of incidents',
+            name: PagerDutyForm.MaxIncidentsKey,
+            description: 'Maximum number of incidents to create based on result set',
+            type: 'string',
+            default: '10',
+            required: true
+        }
+    ]
+
+    static create(json: any): PagerDutyForm | undefined {
+        if (json == null) {
+            return undefined
+        }
+
+        const keys = [PagerDutyForm.ServiceKeyKey,
+        PagerDutyForm.DeDeuplicationIdColumnKey,
+        PagerDutyForm.AutoResolveKey,
+        PagerDutyForm.MaxIncidentsKey
+        ]
+
+        const ok = keys.reduce((p: boolean, key: string) => p && json[key] != null, true)
+
+        if (!ok) return undefined
+
+        return new PagerDutyForm(json[PagerDutyForm.ServiceKeyKey], json[PagerDutyForm.DeDeuplicationIdColumnKey],
+            json[PagerDutyForm.AutoResolveKey],
+            json[PagerDutyForm.MaxIncidentsKey]
+        )
```

Comment:
I'm impressed this compiles. Is there a better/more-idiomatic way to go from "any" to "Maybe PagerDutyForm"?